### PR TITLE
GL example [GL-1]

### PR DESCRIPTION
**Pull request by GabeL-code**
* **Please check if the PR fulfills these requirements**
- [x] The pull request title contains the ticket number in the title. For example [CB-1]

* **What kind of change does this PR introduce?** (Bug fix, new feature...)
      deleted the unused variable. 

* **What is the current behaviour? What does the code do before you made the new changes?** 

    The current behaviour did not change

* **What is the new behavior?**
     no new behavior. 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR? Run a new command?)
  
   none 

* **Other information**:
* none

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            console.log(error);
+            res.json({"api call didnt connect"})
+        })
+})
+
+
+
+
+
+
```
Comment (jorgecan101) [inline index.js]: You could probably get rid of all this blank whitespace here

File: index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    console.log(`ID : ${req.query['id']} removal`)  
+    
+    axios.delete('http://java-sample-api-2020.herokuapp.com/deletePerson?id=', req.query['id'])
+        .then((response) => {
+            console.log(`${response.data}`)
+            res.json({"message":"Successfully deleted. "})
+        })
+        .catch((error) => {
+            console.log(error);
+            res.json({"api call didnt connect"})
```
Comment (arsoljon) [inline index.js]: missing the key for the json response.

File: package.json [hunk 3]
```diff
@@ -11,13 +11,13 @@
...
+    "node": "12.18.3"
   },
   "license": "ISC",
   "dependencies": {
     "axios": "^0.21.1",
     "concurrently": "^6.1.0",
-    "express": "^4.17.1",
-    "cors": "^2.8.5"
+    "cors": "^2.8.5",
+    "express": "^4.17.1"
```
Comment (jorgecan101) [inline package.json]: I don't think we should be modifying the package.json file

File: index.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    console.log(data)
+})
+
+//route to delete a specific person
+
+app.delete('/deletePerson', function(req, res){
+    console.log(`ID : ${req.query['id']} removal`)  
+    
+    axios.delete('http://java-sample-api-2020.herokuapp.com/deletePerson?id=', req.query['id'])
```
Comment (arsoljon) [inline index.js]: On line 45, the delete url is missing the id that is requested. Need to concatenate the 2 strings to have a valid url.

**Review by arsoljon — COMMENTED** [review]
All the routes look good except the delete route. On line 45,  The 'req.query['id] should be concatenated with the url template.

**Comment by jorgecan101** [comment]
Not related to your code itself, but your index.js should be in a folder you created named GL (for your initials). There may be issues if in the current state your index.js gets merged into the one in the main branch rather than in the one in your folder.

File: index.js (hunk 2 above)
Comment (jorgecan101) [reply index.js]: Agree with Jonathan, I get an unexpected token error when trying to run your code.

**Review by jorgecan101 — DISMISSED** [review]
Fix problems based on comments above

File: GL/index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        })
+})
+
+//route to get all people
+app.get('/getAllPeople', function(req, res) {
+    console.log("List of registered users")
+    data = axios.get('http://java-sample-api-2020.herokuapp.com/getAllPeople')
+        .then((response) => {
+            
+            res.json({"message":"Users shown", "Users":`${response.data}`})
```
Comment (jorgecan101) [inline GL/index.js]: In Postman, it continues to display "Users":"[object Object]" instead of what is intended

File: GL/index.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+//route to delete a specific person
+
+app.delete('/deletePerson', function(req, res){
+    console.log(`ID : ${req.query['id']} removal`)  
+    const id = req.query['id']
+    const url = 'http://java-sample-api-2020.herokuapp.com/deletePerson?id='+id
+    
+    axios.delete(url)
+        .then((response) => {
+            console.log(`${response.data}`)
```
Comment (jorgecan101) [inline GL/index.js]: Here it once again displays in the terminal: [object Object] instead of what is intended

**Review by jorgecan101 — COMMENTED** [review]
Should be good apart from the details I commented. Everything else functions correctly.

File: GL/index.js (hunk 6 above)
Comment (Codubee) [reply GL/index.js]: Try just logging the data without the `,$, or {}. I think that is creating the issue.

**Review by Codubee — DISMISSED** [review]
Left a comment

File: package.json [hunk 7]
```diff
@@ -7,7 +7,8 @@
     "test": "echo \"Error: no test specified\" && exit 1",
     "server": "node index.js",
     "client": "cd ../ && npm start --prefix frontend",
-    "dev": "concurrently \"npm run server\" \"npm run client\""
+    "dev": "concurrently \"npm run server\" \"npm run client\"",
+    "start": "node index.js"
```
Comment (arsoljon) [inline package.json]: In the package.json file, on line 11, we do not need the ' "start": "node index.js" ' line as of yet.

File: GL/index.js [hunk 8]
```diff
@@ -46,7 +46,7 @@ app.delete('/deletePerson', function(req, res){
     
     axios.delete(url)
         .then((response) => {
-            console.log(`${response.data}`)
+            console.log(`response.data`)
```
Comment (arsoljon) [inline GL/index.js:49]: on line 49, remove the backticks ` and it will show the intended info.

File: GL/index.js [hunk 9]
```diff
@@ -26,7 +26,7 @@ app.get('/getAllPeople', function(req, res) {
     data = axios.get('http://java-sample-api-2020.herokuapp.com/getAllPeople')
         .then((response) => {
             const data = response.data
```
Comment (arsoljon) [inline GL/index.js]: On line 28, the data variable is not used so consider removing that line.

**Review by arsoljon — COMMENTED** [review]
left some comments.

File: GL/index.js (hunk 8 above)
Comment (jorgecan101) [reply GL/index.js:49]: Agree with Jonathan, right now it displays "response.data" literally instead of the actual data.

File: GL/index.js (hunk 9 above)
Comment (jorgecan101) [reply GL/index.js]: Agree with Jonathan

**Review by arsoljon — APPROVED** [review]
looks good

**Review by jorgecan101 — APPROVED** [review]
looks good

**Pull request merged**